pkg/apis/core/types_shoot.go

```diff
@@ -1205,6 +1208,30 @@ type MaintenanceAutoUpdate struct {
 	MachineImageVersion *bool
 }
 
+// MaintenanceAutoRotation contains information about which rotations should be automatically performed.
+type MaintenanceAutoRotation struct {
+	// Credentials contains information about which credentials should be automatically rotated.
+	Credentials *MaintenanceCredentialsAutoRotation
+}
+
+// MaintenanceCredentialsAutoRotation contains information about which credentials should be automatically rotated.
+type MaintenanceCredentialsAutoRotation struct {
+	// ETCDEncryptionKey configured the automatic rotation for the etcd encryption key.
+	ETCDEncryptionKey *MaintenanceRotationConfig
+	// Observability configured the automatic rotation for the observability credentials.
+	Observability *MaintenanceRotationConfig
+	// SSHKeypair configured the automatic rotation for the ssh keypair for worker nodes.
+	SSHKeypair *MaintenanceRotationConfig
+}
+
+// MaintenanceRotationConfig contains configuration for automatic rotation.
+type MaintenanceRotationConfig struct {
+	// Enabled indicates whether automatic rotation should be performed (default: false).
+	Enabled *bool
```

Comment:
This doesn't have to be a pointer since `MaintenanceRotationConfig` is already a pointer in the `MaintenanceCredentialsAutoRotation` struct right?
```suggestion
	Enabled bool
```

Reply:
I feel like having the `enabled` field as a pointer would help us if at some point we decided to default it for a credential or another type for rotation to `true`. If the user had configured:
``` yaml
spec:
  maintenance:
    autoRotation:
      credentials:
        etcdEncryptionKey:
          rotationPeriod: 68h
```
We would have no way of knowing that the user had not configured the `.enabled` field.